### ISSUE TITLE
Many warnings when compiling with elixir 1.3

### ISSUE THREAD

**Issue by jeanparpaillon** [awetzel/ewebmachine#21]
elixir 1.3 deprecates some languages features used in ewebmachine, producing a lot of warnings

**Comment by jeanparpaillon** [comment]
Most of ewebmachine macros use the (anti-hygienic) `var!` construction in macros for producing side-effects (ie affect context's `conn` and `user_state` variables). 
* PRO: side effect provided through syntaxic sugar, no runtime override
* CON: if used inside a if/cond/... construction, implies unsafe use of variables, which is deprecated

Alternatives are:
1. no side-effect: all functions takes explicitly `conn` and `user_state` variables and returns a copy of them
  * PRO: no side effect
  * CON: heavier syntax, big API change
2. no side effect, with simplified callbacks: all callbacks have a simplified version macro, for instance (erlang syntax but you know...):
```erlang
is_conflict(conn, state) -> {boolean(), conn, state}
is_conflict -> boolean()
```
  * PRO: no side effect, clean syntax
  * CON: (limited) API change: when using simplified macros, `conn` and `state` are not available anymore
3. side effect through process dictionary: original `webmachine` implementation
  * PRO: light syntax
  * CON: performance
4. side effect through external process:
  * each request processing process (P1) is attached another process (Agent ?, ets table ?, P2) storing `conn` and `user_state`
  * P1 process dictionary stores only P2 pid
  * PRO: no API change, clean syntax
  * CON: performance between 1-2 and 3 ?

**Comment by jeanparpaillon** [comment]
Fixed in 7dc41577d29251ce0fb0faa97c9bc2bd5f8266f9
